File: Projeto_6/src/models/UsuarioOperador.cpp
#include "UsuarioOperador.h"
#include <iostream>

// ---------------------------------------------------------
// Implementação da Classe Interna (Nested Class)
// ---------------------------------------------------------
UsuarioOperador::HistoricoAcessos::HistoricoAcessos(std::string recurso, std::string data, int status)
    : recursoAcessado(recurso), dataHora(data), statusCodigo(status) {}

std::string UsuarioOperador::HistoricoAcessos::formatarLog() const {
    return "[" + dataHora + "] Recurso: " + recursoAcessado + " | Status: " + std::to_string(statusCodigo);
}

// ---------------------------------------------------------
// Implementação da Classe Externa (Operador)
// ---------------------------------------------------------
UsuarioOperador::UsuarioOperador(int id, std::string username, std::string senha)
    : UsuarioAutenticavel(id, username, TipoUsuario::OPERADOR), senhaCorreta(senha) {}

bool UsuarioOperador::autenticar(std::string senha) {
    return this->senhaCorreta == senha;
}

void UsuarioOperador::registrarAcesso(std::string recurso, std::string data, int status) {
    acessos.push_back(HistoricoAcessos(recurso, data, status));
}

void UsuarioOperador::gerarRelatorio() const {
    std::cout << "[RELATORIO OPERADOR] Usuario: " << username << " | Historico de Acessos:\n";
    if (acessos.empty()) {
        std::cout << " -> Nenhum acesso registrado.\n";
    } else {
        for (const auto& acesso : acessos) {
            std::cout << " -> " << acesso.formatarLog() << "\n";
        }
    }
}

File: Projeto_6/models/UsuarioOperador.py
from base.UsuarioAutenticavel import UsuarioAutenticavel
from interfaces.Relatorio import Relatorio
from models.TipoUsuario import TipoUsuario

class UsuarioOperador(UsuarioAutenticavel, Relatorio):
    class __HistoricoAcessos:
        def __init__(self):
            self.__registros = []

        def adicionar(self, recurso: str, data_hora: str, status_codigo: int):
            self.__registros.append({
                "recurso_acessado": recurso,
                "data_hora": data_hora,
                "status_codigo": status_codigo
            })

        def obter_todos(self):
            return self.__registros

    def __init__(self, user_id: int, username: str, senha: str):
        super().__init__(user_id, username, senha)
        self.__tipo = TipoUsuario.OPERADOR
        self.__historico = self.__HistoricoAcessos()

    def autenticar(self, senha: str) -> bool:
        return self._verificar_senha(senha)

    def simular_acesso(self, recurso: str, data_hora: str, status: int):
        self.__historico.adicionar(recurso, data_hora, status)

    def gerar_relatorio(self) -> None:
        print(f"\n[RELATÓRIO {self.__tipo.value}] - Operações de Caixa")
        print(f"Colaborador: {self.username}")
        print("Trilha de Auditoria (Acessos):")
        
        registros = self.__historico.obter_todos()
        if not registros:
            print(" - Nenhum acesso registrado.")
        else:
            for reg in registros:
                print(f" -> {reg['data_hora']} | {reg['recurso_acessado']} [Status: {reg['status_codigo']}]")